packages/gatsby/src/schema/types/NodeInterface.js

```diff
@@ -22,13 +21,19 @@ const getOrCreateNodeInterface = schemaComposer => {
       id: `String!`,
       parent: {
         type: `Node`,
-        resolve: async (source, args, context, info) =>
-          context.nodeModel.getNode(source.parent),
+        resolve: (source, args, context, info) => {
+          const { path } = context
+          return context.nodeModel.getNode(source.parent, { path })
+        },
       },
       children: {
         type: `[Node]!`,
-        resolve: async (source, args, context, info) =>
-          source.children.map(context.nodeModel.getNode),
+        resolve: (source, args, context, info) => {
+          const { path } = context
+          return source.children.map(id =>
```

Comment:
The fact that we use this pattern several times, means we need `getNodes` as a model method.